gsi/proxy/proxy_ssl/source/library/proxypolicy.h

```diff
@@ -104,6 +104,9 @@ ASN1_METHOD * PROXYPOLICY_asn1_meth();
 #endif
 
 PROXYPOLICY * PROXYPOLICY_dup(
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
+    const
+#endif
     PROXYPOLICY *                       policy);
```

Comment:
@msalle:
There are additional occurances of non-const `PROXYPOLICY` pointers below. Why don't they get a `const` type qualifier added for OpenSSL 3.0.0?

Reply:
I fixed the ones that gave errors, I think the rest is ok or not coming from OpenSSL?
We probably could make `PROXYPOLICY_print()` and `PROXYPOLICY_print_fp()` use a const, but they are defined in gridcf, not openssl.
Did you have a specific one in mind?

Reply:
No, actually not. I tried to recognize a pattern for this change, but failed, so asked to be sure.

I assumed if `PROXYCERTINFO_dup()` "comes" from OpenSSL, than also the other `PROXYCERTINFO_[...]()` functions come from there and when the `PROXYCERTINFO * [...]` argument is `const` for `PROXYCERTINFO_dup()` with OpenSSL 3.0 it should also be `const` for the other functions where it made sense from their naming.

Now seeing that all but `PROXYCERTINFO_dup()` have an "obvious" implementation in [gsi/proxy/proxy_ssl/source/library/proxycertinfo.c](https://github.com/gridcf/gct/blob/5882044a0eac96b38b276f0ed4f565afc16fe494/gsi/proxy/proxy_ssl/source/library/proxycertinfo.c) makes it a little clearer.